ui/src/hooks/usePolling.ts

```diff
@@ -0,0 +1,110 @@
+import { useRef, useEffect, useCallback, useState } from 'react';
```

Comment:
Instead of implementing our own polling mechanisms, we should look into using an existing library -- e.g. https://swr.vercel.app/docs/getting-started  -- I think that one should implement everything we need for automatic reloads/refreshes.